web-bundle/src/main/java/com/graphhopper/http/GraphHopperManaged.java

```diff
@@ -18,139 +18,117 @@
-            List<JsonFeatureCollection> jsonFeatureCollections = new ArrayList<>();
-            try (DirectoryStream<Path> stream = Files.newDirectoryStream(bordersDirectory, "*.{geojson,json}")) {
-                for (Path borderFile : stream) {
-                    try (BufferedReader reader = Files.newBufferedReader(borderFile, StandardCharsets.UTF_8)) {
-                        JsonFeatureCollection jsonFeatureCollection = localObjectMapper.readValue(reader, JsonFeatureCollection.class);
-                        jsonFeatureCollections.add(jsonFeatureCollection);
-                    }
-                }
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-            SpatialRuleLookupHelper.buildAndInjectCountrySpatialRules(graphHopper,
-                    new Envelope(maxBounds.minLon, maxBounds.maxLon, maxBounds.minLat, maxBounds.maxLat), jsonFeatureCollections);
+            graphHopper = new CustomGraphHopperOSM(configuration);
         }
 
-        ObjectMapper yamlOM = Jackson.initObjectMapper(new ObjectMapper(new YAMLFactory()));
+        String customModelFolder = configuration.getString("custom_model_folder", "");
+        List<Profile> newProfiles = resolveCustomModelFiles(customModelFolder, configuration.getProfiles());
+        configuration.setProfiles(newProfiles);
+
+        // we read all custom speeds mappings into memory so they can be efficiently applied during OSM import. each
+        // custom speed file should be <= 250 MB (nationwide speed mapping file is 235MB)
+        ImmutableMap<String, ImmutableMap<Long, Double>> vehicleNameToCustomSpeeds =
+                CustomSpeedsUtils.getVehicleNameToCustomSpeeds(configuration.getProfiles());
+
+        graphHopper.setEncodedValueFactory(new EncodedValueFactoryWithStableId());
+        graphHopper.setTagParserFactory(new TagParserFactoryWithOsmId());
+        graphHopper.setVehicleTagParserFactory(new ReplicaVehicleTagParserFactory(vehicleNameToCustomSpeeds));
+        graphHopper.setVehicleEncodedValuesFactory(new ReplicaVehicleEncodedValuesFactory(vehicleNameToCustomSpeeds.keySet()));
+        graphHopper.init(configuration);
+        graphHopper.setEncodedValuesString("osmid,stable_id_byte_0,stable_id_byte_1,stable_id_byte_2,stable_id_byte_3,stable_id_byte_4,stable_id_byte_5,stable_id_byte_6,stable_id_byte_7,reverse_stable_id_byte_0,reverse_stable_id_byte_1,reverse_stable_id_byte_2,reverse_stable_id_byte_3,reverse_stable_id_byte_4,reverse_stable_id_byte_5,reverse_stable_id_byte_6,reverse_stable_id_byte_7");
+        graphHopper.setPathDetailsBuilderFactory(new PathDetailsBuilderFactoryWithStableId());
+        graphHopper.setAllowWrites(!Boolean.parseBoolean(System.getenv("GRAPHHOPPER_READ_ONLY")));
+    }
+
+    // TODO: MUST UPDATE EVERY TIME A "GRAPHHOPPER CORE UPDATE" OCCURS
+    // Copied directly from GraphHopperManaged.java on commit f2c5e661308403190545eaec54e8728ab80de11f
```

Comment:
this looks newer than the commit we're depending on, any reason why? and i like that this comment is a reminder for the future, but could we remove the "TODO" here if the work has already been done for this update?

Reply:
Hmm that's a funny oversight, nice catch. Re-copied the function from the correct commit, updated the commit # in the comment, and removed the `TODO` from the comment above it!